### ISSUE TITLE
`typing.Self` not resolving correctly on classmethod property

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2861]
**Describe the bug**
Pyright doesn't resolve the type of `Self` to the current class in a class property return annotation.

**Expected behavior**
The type of `Self` to be `A` in the example code below.

**Screenshots or Code**
```python
from typing_extensions import Self

class A:
    @classmethod
    @property
    def one(cls) -> Self:
        ...

    @classmethod
    @property
    def two(cls) -> type[Self]:
        ...

reveal_type(A.one)  # Type of "A.one" is "Self@A" - Should be "A"
reveal_type(A.two)  # Type of "A.two" is "Type[Self@A]" - Should be "Type[A]"
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version?
Command line (pyright 1.1.208)

**Additional context**
With @Gobot1234's fork of mypy, the revealed types are correct:
```python
main.py:62: note: Revealed type is "def () -> Type[main.A]"
main.py:63: note: Revealed type is "def () -> Type[Self@main.A]"
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.209, which I just published. It will also be included in the next release of pylance.
